lib/conanfile.py

```diff
@@ -43,30 +42,22 @@ def generate(self):
         toolchain_path = path.join(self.build_folder, "conan_toolchain.cmake")
         if path.exists(toolchain_path):
             remove(toolchain_path)
-
-        tc = CMakeToolchain(self)
+        generator = self.conf.get("tools.cmake.cmaketoolchain:generator", default="Ninja Multi-Config")
+        tc = CMakeToolchain(self, generator=generator)
         tc.generate()
 
     def export_sources(self):
-        self.copy("*.h")
-        self.copy("*.cpp")
+        self.copy("hello.h")
+        self.copy("hello.cpp")
         self.copy("CMakeLists.txt")
 
     def build(self):
-        generator = self.conf.get("tools.cmake.cmaketoolchain:generator", default="Ninja Multi-Config")
-        lower_build_type = str(self.settings.build_type).lower()
-
-        env_vars = {}
-
-        # Explicitly add vcvars to the environment. Do not known why is automatically done before calling build() on conan side
-        if self.settings.os == "Windows":
-            env_vars = tools.vcvars_dict(self)
-
-        with tools.environment_append(env_vars):
-            with tools.chdir(self.source_folder):
-                cmd_cmake_config = f"cmake --preset {lower_build_type}" if generator == "Unix Makefiles" else "cmake --preset default"
```

Comment:
This was the command not working. You have to call cmake configure first.
This is the command that `cmake.configure()` does in my machine:
```
----Running------
> "C:\Users\danimtb\.conan\data\hello\1.0.0\_\_\build\f26537b056c9ff213dcc2874372b233408a3e04e\build\generators\conanbuild.bat" && cmake -G "Ninja Multi-Config" -DCMAKE_TOOLCHAIN_FILE="C:/Users/danimtb/.conan/data/hello/1.0.0/_/_/build/f26537b056c9ff213dcc2874372b233408a3e04e/build/generators/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="C:/Users/danimtb/.conan/data/hello/1.0.0/_/_/package/f26537b056c9ff213dcc2874372b233408a3e04e" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" "C:\Users\danimtb\.conan\data\hello\1.0.0\_\_\build\f26537b056c9ff213dcc2874372b233408a3e04e\."
-----------------
**********************************************************************
```
It also applies environment, check the link in the comment below

Reply:
I don't kwno if the presest logic you are doing is needed, as I see the CMake helper is already reading the presets here: https://github.com/conan-io/conan/blob/0fe6058771c7d9585f9fed1935a4e0ec078d3314/conan/tools/cmake/cmake.py#L55
But maybe you want to set and then select your own preset? 😕 no idea